src/py21cmfast/src/HaloBox.c

```diff
@@ -1,105 +1,276 @@
+/* This file contains fucntions for calculating the HaloBox output for 21cmfast, containing the gridded
+ * source properties, either from integrating the conditional mass functions in a cell or from the halo sampler */
+
+//Parameters for the halo box calculations
+//  These are just the values which are calculated at the beginning of ComputeHaloBox and don't change
+//  using this reduces the use of the global parameter structs and allows fewer exp/log unit changes
+struct HaloBoxConstants{
+    double redshift;
+
+    double fstar_10;
+    double alpha_star;
+    double sigma_star;
+
+    double alpha_upper;
+    double pivot_upper;
+    double upper_pivot_ratio;
+
+    double fstar_7;
+    double alpha_star_mini;
+
+    double t_h;
+    double t_star;
+    double sigma_sfr_lim;
+    double sigma_sfr_idx;
+
+    double l_x;
+    double l_x_mini;
+    double sigma_xray;
+
+    double fesc_10;
+    double alpha_esc;
+    double fesc_7;
+
+    double vcb_norel;
+    double mturn_a_nofb;
+    double mturn_m_nofb;
+
+    double Mlim_Fstar;
+    double Mlim_Fesc;
+    double Mlim_Fstar_mini;
+    double Mlim_Fesc_mini;
+};
+
+//struct holding each halo property we currently need.
+//This is only used for both averages over the box/catalogues
+//  as well as an individual halo's properties
+struct HaloProperties{
+    double halo_mass;
+    double stellar_mass;
+    double halo_sfr;
+    double stellar_mini;
```

Comment:
in the spirit of using more explicit names, I think this one is a little cryptic